src/engraving/libmscore/barline.cpp

```diff
@@ -67,21 +67,23 @@ static void undoChangeBarLineType(BarLine* bl, BarLineType barType, bool allStav
     if (!m) {
         return;
     }
-
-    if (barType == BarLineType::START_REPEAT) {
-        m = m->nextMeasure();
-        if (!m) {
-            return;
-        }
-    } else if (bl->barLineType() == BarLineType::START_REPEAT) {
-        if (barType != BarLineType::END_REPEAT) {
-            m->undoChangeProperty(Pid::REPEAT_START, false);
-        }
-        m = m->prevMeasure();
-        if (!m) {
-            return;
+    if (bl->barLineType() == BarLineType::START_REPEAT) {
+        m = m->prevMeasure(); // only type of barline that "belongs" to next measure
+    }
+    if (m->nextMeasure()) {
+        bool isStartRepeat = barType == BarLineType::END_START_REPEAT || barType == BarLineType::START_REPEAT;
+        m->nextMeasure()->undoChangeProperty(Pid::REPEAT_START, isStartRepeat);
+        if (!isStartRepeat) {
+            if (bl->score()->selection().element() == bl) {
+                bl->score()->select(const_cast<BarLine*>(m->endBarLine()));
+            }
+            if (bl->barLineType() == BarLineType::START_REPEAT) {
+                bl->score()->onElementDestruction(bl);
```

Comment:
NotationInteraction still maintains a reference to this otherwise and attempts to draw it until the selection is changed

Reply:
As far as I can tell, this element is not destroyed here, so using onElementDestruction here is incorrect

Reply:
Not deleted as an in-memory object no, but the measure layout code destroys it by removing the owning segment (actually I suspect there's a memory leak, the destructor never seemed to get called, but it no longer belongs to the score).
But without this, the element appears to remain on the screen on the next paint because it's referred to by the NotationInteraction as the "edited object".